sonar-ballerina-plugin/src/main/resources/io/ballerina/sonar/ballerina-3.html

```diff
@@ -0,0 +1,23 @@
+<p>
+    A call to an isolated function is concurrency-safe if it is called with arguments that are safe at least until the call returns.
+    A public function needs to be isolated because it might have been intended to be used in a concurrent environment.
+</p>
+
+<h2>Noncompliant Code Example</h2>
+
+<pre>
+public function helperFunction() {
+   // isolated code
+}
+</pre>
+
+<h2>Compliant Solution</h2>
+
+<p>
+    Mark public non-isolated functions as isolated if they are not the entry point of the program to ensure thread safety.
```

Comment:
```suggestion
    Mark public functions as isolated to ensure the function can be called concurrently.
```

The language guarantees concurrency safety (for the most part). Not marking this as `isolated` doesn't (cannot) make it unsafe.

Reply:
Fixed with f4243e9c45fa8560e64f962cde4abe6c4cf69e01